src/rtos/tasks.c

```diff
@@ -4745,6 +4745,26 @@ uint32_t uxReturn;
 #endif /* configUSE_TASK_NOTIFICATIONS */
 /*-----------------------------------------------------------*/
 
+/*-----------------------------------------------------------*/
+
+#if( configUSE_TASK_NOTIFICATIONS == 1 )
+
+	void task_join(task_t task) {
+		if(!task) return;
+		TaskStatus_t xTaskDetails;
+		do
+		{
+			vTaskGetInfo(task, &xTaskDetails, pdTRUE, E_TASK_STATE_INVALID);
+			if(xTaskDetails.eCurrentState == E_TASK_STATE_DELETED) { 
+				break;
+			}
+		} while (!task_notify_take(true, 20));
```

Comment:
does this mean the caller has to set up the task_notify_when_deleting call before invoking this function? task_join should probably handle that before it goes into the loop